sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,459 @@
+        # it's always true as a new time_idx column is added
+        # to the data before it's passed to underlying model
+        for i in range(output.shape[0]):
+            start_idx = i * max_prediction_length
+            start_time = data.loc[start_idx, time_idx]
+            data.loc[
+                start_idx : start_idx + max_prediction_length - 1, time_idx
+            ] = list(range(start_time, start_time + max_prediction_length))
+
+        # set the instance columns to multi index
+        data.set_index(index_names, inplace=True)
+        self._origin_time_idx.set_index(index_names, inplace=True)
+        # add origin time_idx column to data
+        data = data.join(self._origin_time_idx, on=index_names)
+        # drop _auto_time_idx column
+        data.reset_index(level=list(range(len(index_names))), inplace=True)
+        data.drop("_auto_time_idx", axis=1, inplace=True)
+        index_names.remove("_auto_time_idx")
+        # drop _auto_group_id column
+        if self._index_len == 1:
+            data.drop("_auto_group_id", axis=1, inplace=True)
+            index_names.remove("_auto_group_id")
+        # reindex to origin multiindex
+        # the last of self._new_index_names is the time index
+        data.set_index(
+            index_names + [self._new_index_names[-1]],
+            inplace=True,
+        )
+        # set index names back to original input in fit
+        if self._index_len == 1:
+            rename_input = self._index_names[0]
+        else:
+            rename_input = self._index_names
+        data.index.rename(rename_input, inplace=True)
+        # set target name back to original input in fit
+        data.columns = [self._target_name]
+        # convert back to pd.series if needed
+        if self._convert_to_series:
+            data = pd.Series(
+                data=data[self._target_name], index=data.index, name=self._target_name
+            )
+        return data
+
+    def _dummy_X(self, X, y):
+        rX = self.algorithm_class.__name__ == "TemporalFusionTransformer"
+        if rX and X is None:
+            print(  # noqa T001
+                "TemporalFusionTransformer requires X!\
+                A constant dummy X with values all zero will be used!"
+            )
+            X = pd.DataFrame(data=np.zeros(len(y)), index=y.index)
+        return X
+
+    def _extend_y(self, y: pd.DataFrame, fh):
+        index = fh.to_absolute_index(self.cutoff)
+        _y = pd.DataFrame(index=index, columns=y.columns)
+        _y.index.rename(y.index.names[-1], inplace=True)
+        _y.fillna(0, inplace=True)
+        len_levels = len(y.index.names)
+        if len_levels == 1:
+            _y = pd.concat([y, _y])
+        else:
+            _y = y.groupby(level=list(range(len_levels - 1))).apply(
+                lambda x: pd.concat([x.droplevel(list(range(len_levels - 1))), _y])
+            )
+        return _y
+
+
+def _series_to_frame(data):
```

Comment:
@fkiraly do we have already util methods for this? I suppose that such methods are required often in Global forecasting.

Reply:
For now it is okay so. Such refinements could be made in additional PRs.